compute_endpoint/globus_compute_endpoint/endpoint/config/config.py

```diff
@@ -502,24 +502,19 @@ def identity_mapping_config_path(self) -> pathlib.Path | None:
 
     @identity_mapping_config_path.setter
     def identity_mapping_config_path(self, val: os.PathLike | str | None):
-        self._identity_mapping_config_path: pathlib.Path | None
-        if is_privileged():
-            if not val:
-                raise ValueError(
-                    "Identity mapping required.  (Hint: identity_mapping_config_path)"
-                )
-
-            _p = pathlib.Path(val)
-            if not _p.exists():
-                raise ValueError(f"Identity mapping config path not found ({_p})")
-            self._identity_mapping_config_path = _p
-        else:
-            self._identity_mapping_config_path = None
-            if val:
-                log.warning(
-                    "Identity mapping specified, but process is not privileged;"
-                    " ignoring identity mapping configuration."
-                )
+        self._identity_mapping_config_path: pathlib.Path | None = None
+        if not val:
+            return
+        if not is_privileged():
+            log.warning(
+                "Identity mapping specified, but process is not privileged;"
+                " ignoring identity mapping configuration."
+            )
+            return
```

Comment:
I know it came before, but my thinking has evolved.  Other than the error in the next if-condition, that implements the "fail early and fast" methodology, I'm thinking that the config does not need to be responsible for warning the user if the privileges are not correct.  It's only if the logic works out to try _using_ the mapping by a non-privileged process that we should warn.

Reply:
Removed